test/gui/shared/steps/steps.py

```diff
@@ -391,15 +389,23 @@ def step(context):
     clickTab(waitForObject(names.stack_QTabWidget), "Not Synced")
 
 
-def openPublicLinkDialog(context, resource):
+def openSharingDialog(context, resource, fileType='file'):
     resource = sanitizePath(substituteInLineCodes(context, resource))
-    waitFor(lambda: isFileSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    
+    if fileType == 'folder':
+        waitFor(lambda: isFolderSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    else:
+        waitFor(lambda: isFileSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+            
     waitFor(lambda: shareResource(resource), context.userData['clientSyncTimeout'] * 1000)
+    
+def openPublicLinkDialog(context, resource, fileType):
+    openSharingDialog(context, resource, fileType)
     mouseClick(waitForObject(names.qt_tabwidget_tabbar_Public_Links_TabItem), 0, 0, Qt.NoModifier, Qt.LeftButton)
 
 @When('the user opens the public links dialog of "|any|" using the client-UI')
 def step(context, resource):
-    openPublicLinkDialog(context,resource)
+    openPublicLinkDialog(context,resource,'')
```

Comment:
```suggestion
    openPublicLinkDialog(context,resource)
```
or 
`openPublicLinkDialog(context,resource, 'file')`

same in other places

Reply:
fixed